components/clp-package-utils/clp_package_utils/scripts/native/compress.py

```
@@ -153,41 +156,152 @@ def _generate_clp_io_config(
+    """
+    Parse and validate S3 object URLs.
+
+    Validates:
+    - All URLs have same region and bucket.
+    - No duplicate keys.
+    - For multiple URLs: non-empty common prefix exists.
+
+    :param urls: List of S3 URLs
+    :return: Tuple of (region_code, bucket, key_prefix, keys).
+             - For single URL: key_prefix is the full key, keys is None (more efficient).
+             - For multiple URLs: key_prefix is the common prefix, keys are full keys.
+    :raises ValueError: If validation fails.
+    """
+    if len(urls) == 0:
+        raise ValueError("No URLs provided")
+
+    parsed_urls = []
+    for url in urls:
+        try:
+            region_code, bucket, key = parse_s3_url(url)
+            parsed_urls.append((region_code, bucket, key))
+        except ValueError as e:
+            raise ValueError(f"Failed to parse URL '{url}': {e}")
+
+    first_region, first_bucket, _ = parsed_urls[0]
+    for region, bucket, _ in parsed_urls:
+        if region != first_region:
+            raise ValueError(
+                f"All URLs must have same region code. Found '{first_region}' and '{region}'"
+            )
+        if bucket != first_bucket:
+            raise ValueError(
+                f"All URLs must have same bucket. Found '{first_bucket}' and '{bucket}'"
+            )
+
+    keys = [key for _, _, key in parsed_urls]
+
+    if len(keys) != len(set(keys)):
+        raise ValueError("Duplicate keys found in URLs")
+
+    # For a single key, use the full key as prefix with keys=None
+    # For multiple keys, find common prefix and provide explicit keys list
+    if len(keys) == 1:
+        key_prefix = keys[0]
+        keys = None
+    else:
+        key_prefix = os.path.commonprefix(keys)
+        if not key_prefix:
+            raise ValueError("No common prefix found among object keys")
+
+    return first_region, first_bucket, key_prefix, keys
+
+
+def _parse_s3_key_prefix_url(url: str) -> Tuple[str, str, str]:
```

Do we need this? `parse_s3_url` should throw a similar error already, right?